### PR TITLE
Structured log migration for pkg/probe

### PR DESCRIPTION
**Pull request by jialaijun**
in pkg/probe/exec/exec.go
in pkg/probe/http/http.go
in pkg/probe/tcp/tcp.go

**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it:**

Ref:

- [keps/sig-instrumentation/1602-structured-logging](https://github.com/kubernetes/enhancements/tree/master/keps/sig-instrumentation/1602-structured-logging)

- [Structured Logging migration instructions](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-instrumentation/migration-to-structured-logging.md)

**Does this PR introduce a user-facing change?:**

`Migrate some log messages to structured logging `

### CODE REVIEW

File: pkg/probe/tcp/tcp.go [hunk 1]
```diff
@@ -55,7 +56,7 @@ func DoTCPProbe(addr string, timeout time.Duration) (probe.Result, string, error
 	}
 	err = conn.Close()
 	if err != nil {
-		klog.Errorf("Unexpected error closing TCP probe socket: %v (%#v)", err, err)
+		klog.ErrorS(fmt.Errorf("%v (%#v)", err, err), "Unexpected error closing TCP probe socket")
```
Comment (ehashman) [inline pkg/probe/tcp/tcp.go:59]:
```suggestion
		klog.ErrorS(err, "Unexpected error closing TCP probe socket")
```

**Review by ehashman — CHANGES_REQUESTED** [review]
/triage accepted
/priority important-soon
/ok-to-test

Please fix your release note. It needs to begin with ```release-note

**Comment by ehashman** [comment]
/remove-priority important-soon
/priority backlog

Since this is against pkg/probe we didn't commit to migrating it this release, I'm going to remove this from the milestone for now (although it'd be great if we could get this merged)

**Comment by pacoxu** [comment]
subset of #100834 which was reviewed.

**Pull request closed without merging**